mesos_master/datadog_checks/mesos_master/mesos_master.py

```diff
@@ -178,7 +178,21 @@ def _get_json(self, url, timeout, verify=True, tags=None):
         return r.json()
 
     def _get_master_state(self, url, timeout, verify, tags):
-        return self._get_json(url + '/state.json', timeout, verify, tags)
+        try:
+            # version < 1.8.0
+            endpoint = '/state.json'
+            master_state = self._get_json(url + endpoint, timeout, verify, tags)
+            if master_state is not None:
+                self.version = list(map(int, master_state['version'].split('.')))
+        except Exception as e:
+            msg = str(e)
+            self.log.warning('Encounted error getting state at {}{}, message: {}'.format(url, endpoint, msg))
```

Comment:
nit: we could concatenate url and endpoint together before, as the same concatenation happens two lines after.
nit: you can do something like
```python
self.log.warning('String parameters %s, %s and %s', '1', '2', '3')
```
to do string formatting only if the message is logged.

Reply:
https://github.com/DataDog/integrations-core/blob/a999f91c1141ac92256ffc7237f6ccc60e0d8789/mesos_master/datadog_checks/mesos_master/mesos_master.py#L188-L189

I made the concatenation, `endpoint = url  + '/endpoint'`, and passed the variable `endpoint` to `get_json()`. 

Stylistically, is it better to reassign the concatenation to the variable, `url`, and pass that to `get_json()` to preserve the parameter naming convention? I.e. `url += '/endpoint'`.

Reply:
I'd say it's mostly about if the variable name still reflect what it's for or not and if you need to reuse it later or not.
My point was about preventing code duplication, it's nice this way 👍